### ISSUE TITLE
Fatal error when trying to show datepicker on iOS with Compose Multiplatform

### ISSUE THREAD

**Issue by maltsev-gorskij** [Syer10/compose-material-dialogs#2]
**Describe the bug**
I am trying simple example for displaying datePicker MaterialDialog with this manual and got fatal on iOS 16.4 simulator. Android build working well
[https://vanpra.github.io/compose-material-dialogs/DateTimePicker/](https://vanpra.github.io/compose-material-dialogs/DateTimePicker/)

**To Reproduce**
Minimal reprodusable example.

Just pass true to shouldShow

```
@Composable
fun DatePickerManager(shouldShow: Boolean) {
    val datePickerState = rememberMaterialDialogState()

    LaunchedEffect(shouldShow) {
        if(shouldShow) datePickerState.show()
    }

    DatePicker(datePickerState) {
        // TODO -- do something
    }
}

@Composable
private fun DatePicker(
    datePickerState: MaterialDialogState,
    onDateSelect: (LocalDate) -> Unit
) {
    MaterialDialog(
        dialogState = datePickerState,
        buttons = {
            positiveButton("OK")
            negativeButton("Cancel")
        }
    ) {
        datepicker() { date -> onDateSelect(date) }
    }
}
```

**Expected behaviour**
Should appear on iOS as on Android

**Additional context**
This is what i got in Android Studio Flamingo | 2022.2.1 Patch 2 run log

```
/Users/Lyriandr/Documents/GitRepos/Solution/ConferenceAppKMM/build/ios/Debug-iphonesimulator/iosApp.app
Uncaught Kotlin exception: kotlin.ArrayIndexOutOfBoundsException
    at 0   iosApp                              0x100bd21b3        kfun:kotlin.Throwable#<init>(){} + 99 
    at 1   iosApp                              0x100bca213        kfun:kotlin.Exception#<init>(){} + 91 
    at 2   iosApp                              0x100bca657        kfun:kotlin.RuntimeException#<init>(){} + 91 
    at 3   iosApp                              0x100bca923        kfun:kotlin.IndexOutOfBoundsException#<init>(){} + 91 
    at 4   iosApp                              0x100bcb553        kfun:kotlin.ArrayIndexOutOfBoundsException#<init>(){} + 91 
    at 5   iosApp                              0x100c060f7        ThrowArrayIndexOutOfBoundsException + 131 
    at 6   iosApp                              0x101dd827b        Kotlin_Array_get + 43 
    at 7   iosApp                              0x101a5d117        kfun:com.vanpra.composematerialdialogs.datetime.util#plus__at__kotlinx.datetime.DayOfWeek(kotlin.Long){}kotlinx.datetime.DayOfWeek + 375 
    at 8   iosApp                              0x101a4146b        kfun:com.vanpra.composematerialdialogs.datetime.date.DayOfWeekHeader#internal + 2835 
    at 9   iosApp                              0x101a3b007        kfun:com.vanpra.composematerialdialogs.datetime.date.CalendarView#internal + 7731 
    at 10  iosApp                              0x101a4ae7f        kfun:com.vanpra.composematerialdialogs.datetime.date.DatePickerImpl$lambda$8#internal + 12819 
    at 11  iosApp                              0x101a54283        kfun:com.vanpra.composematerialdialogs.datetime.date.$DatePickerImpl$lambda$8$FUNCTION_REFERENCE$5.invoke#internal + 183 
    at 12  iosApp                              0x101a5453b        kfun:com.vanpra.composematerialdialogs.datetime.date.$DatePickerImpl$lambda$8$FUNCTION_REFERENCE$5.$<bridge-UNNBNB>invoke(kotlin.Int;androidx.compose.runtime.Composer;kotlin.Int){}#internal + 227 
    at 13  iosApp                              0x100f581fb        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#invoke(kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any? + 927 
    at 14  iosApp                              0x100f6a0c7        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#$<bridge-NNNNNB>invoke(kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any?(kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Any?){}kotlin.Any? + 223 
    at 15  iosApp                              0x101601ab3        kfun:androidx.compose.foundation.pager.Pager$lambda$7$lambda$6$lambda$5#internal + 6431 
    at 16  iosApp                              0x101609c3f        kfun:androidx.compose.foundation.pager.$Pager$lambda$7$lambda$6$lambda$5$FUNCTION_REFERENCE$8.invoke#internal + 199 
    at 17  iosApp                              0x101609eaf        kfun:androidx.compose.foundation.pager.$Pager$lambda$7$lambda$6$lambda$5$FUNCTION_REFERENCE$8.$<bridge-UNNNBNB>invoke(androidx.compose.foundation.lazy.LazyItemScope;kotlin.Int;androidx.compose.runtime.Composer;kotlin.Int){}#internal + 263 
    at 18  iosApp                              0x100f5886f        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#invoke(kotlin.Any?;kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any? + 971 
    at 19  iosApp                              0x100f6a1ff        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#$<bridge-NNNNNNB>invoke(kotlin.Any?;kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any?(kotlin.Any?;kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Any?){}kotlin.Any? + 259 
    at 20  iosApp                              0x1015685b3        kfun:androidx.compose.foundation.lazy.LazyListItemProviderImpl.<init>$lambda$1$lambda$0#internal + 887 
    at 21  iosApp                              0x101569693        kfun:androidx.compose.foundation.lazy.LazyListItemProviderImpl.$<init>$lambda$1$lambda$0$FUNCTION_REFERENCE$6.invoke#internal + 147 
    at 22  iosApp                              0x10156985b        kfun:androidx.compose.foundation.lazy.LazyListItemProviderImpl.$<init>$lambda$1$lambda$0$FUNCTION_REFERENCE$6.$<bridge-UNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}#internal + 183 
    at 23  iosApp                              0x100f57bc3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any? + 875 
    at 24  iosApp                              0x100f69fb3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#$<bridge-NNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any?(androidx.compose.runtime.Composer;kotlin.Any?){}kotlin.Any? + 187 
    at 25  iosApp                              0x100e6dc13        kfun:androidx.compose.runtime#CompositionLocalProvider(kotlin.Array<out|androidx.compose.runtime.ProvidedValue<*>>...;kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>;androidx.compose.runtime.Composer?;kotlin.Int){} + 819 
    at 26  iosApp                              0x1015df6eb        kfun:androidx.compose.foundation.lazy.layout#LazyLayoutPinnableItem(kotlin.Any?;kotlin.Int;androidx.compose.foundation.lazy.layout.LazyLayoutPinnedItemList;kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>;androidx.compose.runtime.Composer?;kotlin.Int){} + 5463 
    at 27  iosApp                              0x101568ecf        kfun:androidx.compose.foundation.lazy.LazyListItemProviderImpl.<init>$lambda$1#internal + 2015 
    at 28  iosApp                              0x101569167        kfun:androidx.compose.foundation.lazy.LazyListItemProviderImpl.$<init>$lambda$1$FUNCTION_REFERENCE$4.invoke#internal + 187 
    at 29  iosApp                              0x101569363        kfun:androidx.compose.foundation.lazy.LazyListItemProviderImpl.$<init>$lambda$1$FUNCTION_REFERENCE$4.$<bridge-UNNNBNB>invoke(androidx.compose.foundation.lazy.layout.IntervalList.Interval<androidx.compose.foundation.lazy.LazyListIntervalContent>;kotlin.Int;androidx.compose.runtime.Composer;kotlin.Int){}#internal + 263 
    at 30  iosApp                              0x100f5886f        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#invoke(kotlin.Any?;kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any? + 971 
    at 31  iosApp                              0x100f6a1ff        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#$<bridge-NNNNNNB>invoke(kotlin.Any?;kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any?(kotlin.Any?;kotlin.Any?;androidx.compose.runtime.Composer;kotlin.Any?){}kotlin.Any? + 259 
    at 32  iosApp                              0x1015d9063        kfun:androidx.compose.foundation.lazy.layout.DefaultLazyLayoutItemsProvider.Item#internal + 1875 
    at 33  iosApp                              0x101567c8b        kfun:androidx.compose.foundation.lazy.LazyListItemProviderImpl.Item#internal + 1663 
    at 34  iosApp                              0x1015d775f        kfun:androidx.compose.foundation.lazy.layout.DefaultDelegatingLazyLayoutItemProvider.Item#internal + 1763 
    at 35  iosApp                              0x10156a347        kfun:androidx.compose.foundation.lazy.object-1.Item#internal + 575 
    at 36  iosApp                              0x1015d4a67        kfun:androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory.CachedItemContent.createContentLambda$lambda$2$lambda$0#internal + 763 
    at 37  iosApp                              0x1015d5cb3        kfun:androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory.CachedItemContent.$createContentLambda$lambda$2$lambda$0$FUNCTION_REFERENCE$3.invoke#internal + 143 
    at 38  iosApp                              0x1015d5e3b        kfun:androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory.CachedItemContent.$createContentLambda$lambda$2$lambda$0$FUNCTION_REFERENCE$3.$<bridge-UNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}#internal + 183 
    at 39  iosApp                              0x100f57bc3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any? + 875 
    at 40  iosApp                              0x100f69fb3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#$<bridge-NNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any?(androidx.compose.runtime.Composer;kotlin.Any?){}kotlin.Any? + 187 
    at 41  iosApp                              0x100e6dc13        kfun:androidx.compose.runtime#CompositionLocalProvider(kotlin.Array<out|androidx.compose.runtime.ProvidedValue<*>>...;kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>;androidx.compose.runtime.Composer?;kotlin.Int){} + 819 
    at 42  iosApp                              0x1010f9eb3        kfun:androidx.compose.runtime.saveable.SaveableStateHolderImpl.SaveableStateProvider#internal + 3143 
    at 43  iosApp                              0x1015ee93f        kfun:androidx.compose.foundation.lazy.layout.LazySaveableStateHolder.SaveableStateProvider#internal + 2335 
    at 44  iosApp                              0x1015d54e7        kfun:androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory.CachedItemContent.createContentLambda$lambda$2#internal + 2171 
    at 45  iosApp                              0x1015d5a43        kfun:androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory.CachedItemContent.$createContentLambda$lambda$2$FUNCTION_REFERENCE$2.invoke#internal + 143 
    at 46  iosApp                              0x1015d5bef        kfun:androidx.compose.foundation.lazy.layout.LazyLayoutItemContentFactory.CachedItemContent.$createContentLambda$lambda$2$FUNCTION_REFERENCE$2.$<bridge-UNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}#internal + 183 
    at 47  iosApp                              0x100f57bc3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any? + 875 
    at 48  iosApp                              0x100f69fb3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#$<bridge-NNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any?(androidx.compose.runtime.Composer;kotlin.Any?){}kotlin.Any? + 187 
    at 49  iosApp                              0x10128148b        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.subcompose$lambda$0#internal + 1207 
    at 50  iosApp                              0x1012823f7        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.$subcompose$lambda$0$FUNCTION_REFERENCE$2.invoke#internal + 143 
    at 51  iosApp                              0x1012825a3        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.$subcompose$lambda$0$FUNCTION_REFERENCE$2.$<bridge-UNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}#internal + 183 
    at 52  iosApp                              0x100f57bc3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any? + 875 
    at 53  iosApp                              0x100f69fb3        kfun:androidx.compose.runtime.internal.ComposableLambdaImpl#$<bridge-NNNNB>invoke(androidx.compose.runtime.Composer;kotlin.Int){}kotlin.Any?(androidx.compose.runtime.Composer;kotlin.Any?){}kotlin.Any? + 187 
    at 54  iosApp                              0x100f564ab        kfun:androidx.compose.runtime#invokeComposable(androidx.compose.runtime.Composer;kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>){} + 339 
    at 55  iosApp                              0x100e497fb        kfun:androidx.compose.runtime.ComposerImpl.doCompose$lambda$27#internal + 303 
    at 56  iosApp                              0x100e4f56f        kfun:androidx.compose.runtime.ComposerImpl.$doCompose$lambda$27$FUNCTION_REFERENCE$33.invoke#internal + 103 
    at 57  iosApp                              0x100e4f703        kfun:androidx.compose.runtime.ComposerImpl.$doCompose$lambda$27$FUNCTION_REFERENCE$33.$<bridge-UNN>invoke(){}#internal + 95 
    at 58  iosApp                              0x100e7199b        kfun:androidx.compose.runtime#observeDerivedStateRecalculations(kotlin.Function1<androidx.compose.runtime.State<*>,kotlin.Unit>;kotlin.Function1<androidx.compose.runtime.State<*>,kotlin.Unit>;kotlin.Function0<0:0>){0§<kotlin.Any?>} + 1015 
    at 59  iosApp                              0x100e3d267        kfun:androidx.compose.runtime.ComposerImpl.doCompose#internal + 2247 
    at 60  iosApp                              0x100e3c233        kfun:androidx.compose.runtime.ComposerImpl#composeContent(androidx.compose.runtime.collection.IdentityArrayMap<androidx.compose.runtime.RecomposeScopeImpl,androidx.compose.runtime.collection.IdentityArraySet<kotlin.Any>?>;kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>){} + 431 
    at 61  iosApp                              0x100e5e9b7        kfun:androidx.compose.runtime.CompositionImpl#composeContent(kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>){} + 535 
    at 62  iosApp                              0x100e8a6bf        kfun:androidx.compose.runtime.Recomposer#composeInitial(androidx.compose.runtime.ControlledComposition;kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>){} + 911 
    at 63  iosApp                              0x100e42fdf        kfun:androidx.compose.runtime.ComposerImpl.CompositionContextImpl.composeInitial#internal + 243 
    at 64  iosApp                              0x100e42fdf        kfun:androidx.compose.runtime.ComposerImpl.CompositionContextImpl.composeInitial#internal + 243 
    at 65  iosApp                              0x100e42fdf        kfun:androidx.compose.runtime.ComposerImpl.CompositionContextImpl.composeInitial#internal + 243 
    at 66  iosApp                              0x100e42fdf        kfun:androidx.compose.runtime.ComposerImpl.CompositionContextImpl.composeInitial#internal + 243 
    at 67  iosApp                              0x100e42fdf        kfun:androidx.compose.runtime.ComposerImpl.CompositionContextImpl.composeInitial#internal + 243 
    at 68  iosApp                              0x100e42fdf        kfun:androidx.compose.runtime.ComposerImpl.CompositionContextImpl.composeInitial#internal + 243 
    at 69  iosApp                              0x100e5deb7        kfun:androidx.compose.runtime.CompositionImpl#setContent(kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>){} + 479 
    at 70  iosApp                              0x10127c74f        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.subcomposeInto#internal + 595 
    at 71  iosApp                              0x10127c347        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.subcompose#internal.249 + 907 
    at 72  iosApp                              0x10127bf37        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.subcompose#internal + 1287 
    at 73  iosApp                              0x10127b94b        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState#subcompose(kotlin.Any?;kotlin.Function2<androidx.compose.runtime.Composer,kotlin.Int,kotlin.Unit>){}kotlin.collections.List<androidx.compose.ui.layout.Measurable> + 2715 
    at 74  iosApp                              0x101280f9f        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.Scope.subcompose#internal + 159 
    at 75  iosApp                              0x1015dbc9b        kfun:androidx.compose.foundation.lazy.layout.LazyLayoutMeasureScopeImpl#measure(kotlin.Int;androidx.compose.ui.unit.Constraints){}kotlin.collections.List<androidx.compose.ui.layout.Placeable> + 919 
    at 76  iosApp                              0x101585f4f        kfun:androidx.compose.foundation.lazy.LazyMeasuredItemProvider#getAndMeasure(androidx.compose.foundation.lazy.DataIndex){}androidx.compose.foundation.lazy.LazyMeasuredItem + 483 
    at 77  iosApp                              0x10156c1b3        kfun:androidx.compose.foundation.lazy#measureLazyList(kotlin.Int;androidx.compose.foundation.lazy.LazyListItemProvider;androidx.compose.foundation.lazy.LazyMeasuredItemProvider;kotlin.Int;kotlin.Int;kotlin.Int;kotlin.Int;androidx.compose.foundation.lazy.DataIndex;kotlin.Int;kotlin.Float;androidx.compose.ui.unit.Constraints;kotlin.Boolean;kotlin.collections.List<kotlin.Int>;androidx.compose.foundation.layout.Arrangement.Vertical?;androidx.compose.foundation.layout.Arrangement.Horizontal?;kotlin.Boolean;andr + 3539 
    at 78  iosApp                              0x1015563db        kfun:androidx.compose.foundation.lazy.rememberLazyListMeasurePolicy$lambda$4#internal + 5991 
    at 79  iosApp                              0x101556f97        kfun:androidx.compose.foundation.lazy.$rememberLazyListMeasurePolicy$lambda$4$FUNCTION_REFERENCE$2.invoke#internal + 227 
    at 80  iosApp                              0x101557397        kfun:androidx.compose.foundation.lazy.$rememberLazyListMeasurePolicy$lambda$4$FUNCTION_REFERENCE$2.$<bridge-NNNNB>invoke(androidx.compose.foundation.lazy.layout.LazyLayoutMeasureScope;androidx.compose.ui.unit.Constraints){}androidx.compose.foundation.lazy.LazyListMeasureResult#internal + 187 
    at 81  iosApp                              0x1015cddfb        kfun:androidx.compose.foundation.lazy.layout.LazyLayout$lambda$2$lambda$1#internal + 415 
    at 82  iosApp                              0x1015d01cf        kfun:androidx.compose.foundation.lazy.layout.$LazyLayout$lambda$2$lambda$1$FUNCTION_REFERENCE$3.invoke#internal + 151 
    at 83  iosApp                              0x1015d039f        kfun:androidx.compose.foundation.lazy.layout.$LazyLayout$lambda$2$lambda$1$FUNCTION_REFERENCE$3.$<bridge-NNNNB>invoke(androidx.compose.ui.layout.SubcomposeMeasureScope;androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.MeasureResult#internal + 187 
    at 84  iosApp                              0x101281c27        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.object-1.measure#internal + 799 
    at 85  iosApp                              0x1012a0a57        kfun:androidx.compose.ui.node.InnerNodeCoordinator#measure(androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.Placeable + 959 
    at 86  iosApp                              0x10120118b        kfun:androidx.compose.ui.graphics.SimpleGraphicsLayerModifier.measure#internal + 307 
    at 87  iosApp                              0x1012aa7d3        kfun:androidx.compose.ui.node.LayoutModifierNodeCoordinator#measure(androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.Placeable + 467 
    at 88  iosApp                              0x1012da1c7        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure$lambda$0#internal + 203 
    at 89  iosApp                              0x1012da393        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.$performMeasure$lambda$0$FUNCTION_REFERENCE$12.invoke#internal + 99 
    at 90  iosApp                              0x1012da4c3        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.$performMeasure$lambda$0$FUNCTION_REFERENCE$12.$<bridge-UNN>invoke(){}#internal + 95 
    at 91  iosApp                              0x100f14d2b        kfun:androidx.compose.runtime.snapshots.Snapshot.Companion#observe(kotlin.Function1<kotlin.Any,kotlin.Unit>?;kotlin.Function1<kotlin.Any,kotlin.Unit>?;kotlin.Function0<0:0>){0§<kotlin.Any?>}0:0 + 1147 
    at 92  iosApp                              0x100f53b03        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads$lambda$3#internal + 235 
    at 93  iosApp                              0x100f5412b        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver.$observeReads$lambda$3$FUNCTION_REFERENCE$3.invoke#internal + 99 
    at 94  iosApp                              0x100f5427f        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver.$observeReads$lambda$3$FUNCTION_REFERENCE$3.$<bridge-UNN>invoke(){}#internal + 95 
    at 95  iosApp                              0x100e7199b        kfun:androidx.compose.runtime#observeDerivedStateRecalculations(kotlin.Function1<androidx.compose.runtime.State<*>,kotlin.Unit>;kotlin.Function1<androidx.compose.runtime.State<*>,kotlin.Unit>;kotlin.Function0<0:0>){0§<kotlin.Any?>} + 1015 
    at 96  iosApp                              0x100f4e933        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver#observeReads(0:0;kotlin.Function1<0:0,kotlin.Unit>;kotlin.Function0<kotlin.Unit>){0§<kotlin.Any>} + 1155 
    at 97  iosApp                              0x101311ac3        kfun:androidx.compose.ui.node.OwnerSnapshotObserver#observeReads(0:0;kotlin.Function1<0:0,kotlin.Unit>;kotlin.Function0<kotlin.Unit>){0§<androidx.compose.ui.node.OwnerScope>} + 247 
    at 98  iosApp                              0x1013118a7        kfun:androidx.compose.ui.node.OwnerSnapshotObserver#observeMeasureSnapshotReads(androidx.compose.ui.node.LayoutNode;kotlin.Boolean;kotlin.Function0<kotlin.Unit>){} + 443 
    at 99  iosApp                              0x1012d93a7        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure#internal + 803 
    at 100 iosApp                              0x1012cd2e7        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.MeasurePassDelegate#remeasure(androidx.compose.ui.unit.Constraints){}kotlin.Boolean + 927 
    at 101 iosApp                              0x1012cceff        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.MeasurePassDelegate#measure(androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.Placeable + 751 
    at 102 iosApp                              0x10140890f        kfun:androidx.compose.foundation.layout.boxMeasurePolicy$lambda$6#internal + 1399 
    at 103 iosApp                              0x101409d6b        kfun:androidx.compose.foundation.layout.$boxMeasurePolicy$lambda$6$FUNCTION_REFERENCE$2.measure#internal + 243 
    at 104 iosApp                              0x101409f6b        kfun:androidx.compose.foundation.layout.$boxMeasurePolicy$lambda$6$FUNCTION_REFERENCE$2.$<bridge-NNNNU>measure@androidx.compose.ui.layout.MeasureScope(androidx.compose.ui.layout.MeasureScope;kotlin.collections.List<androidx.compose.ui.layout.Measurable>){}androidx.compose.ui.layout.MeasureResult#internal + 271 
    at 105 iosApp                              0x10140dcab        kfun:androidx.compose.foundation.layout.BoxWithConstraints$lambda$1#internal + 695 
    at 106 iosApp                              0x10140e017        kfun:androidx.compose.foundation.layout.$BoxWithConstraints$lambda$1$FUNCTION_REFERENCE$0.invoke#internal + 155 
    at 107 iosApp                              0x10140e227        kfun:androidx.compose.foundation.layout.$BoxWithConstraints$lambda$1$FUNCTION_REFERENCE$0.$<bridge-NNNNB>invoke(androidx.compose.ui.layout.SubcomposeMeasureScope;androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.MeasureResult#internal + 187 
    at 108 iosApp                              0x101281c27        kfun:androidx.compose.ui.layout.LayoutNodeSubcompositionsState.object-1.measure#internal + 799 
    at 109 iosApp                              0x1012a0a57        kfun:androidx.compose.ui.node.InnerNodeCoordinator#measure(androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.Placeable + 959 
    at 110 iosApp                              0x101445773        kfun:androidx.compose.foundation.layout.SizeModifier.measure#internal + 1247 
    at 111 iosApp                              0x1012928a7        kfun:androidx.compose.ui.node.BackwardsCompatNode#measure__at__androidx.compose.ui.layout.MeasureScope(androidx.compose.ui.layout.Measurable;androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.MeasureResult + 411 
    at 112 iosApp                              0x1012aa7d3        kfun:androidx.compose.ui.node.LayoutModifierNodeCoordinator#measure(androidx.compose.ui.unit.Constraints){}androidx.compose.ui.layout.Placeable + 467 
    at 113 iosApp                              0x1012da1c7        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure$lambda$0#internal + 203 
    at 114 iosApp                              0x1012da393        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.$performMeasure$lambda$0$FUNCTION_REFERENCE$12.invoke#internal + 99 
    at 115 iosApp                              0x1012da4c3        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.$performMeasure$lambda$0$FUNCTION_REFERENCE$12.$<bridge-UNN>invoke(){}#internal + 95 
    at 116 iosApp                              0x100f14d2b        kfun:androidx.compose.runtime.snapshots.Snapshot.Companion#observe(kotlin.Function1<kotlin.Any,kotlin.Unit>?;kotlin.Function1<kotlin.Any,kotlin.Unit>?;kotlin.Function0<0:0>){0§<kotlin.Any?>}0:0 + 1147 
    at 117 iosApp                              0x100f53b03        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver.observeReads$lambda$3#internal + 235 
    at 118 iosApp                              0x100f5412b        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver.$observeReads$lambda$3$FUNCTION_REFERENCE$3.invoke#internal + 99 
    at 119 iosApp                              0x100f5427f        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver.$observeReads$lambda$3$FUNCTION_REFERENCE$3.$<bridge-UNN>invoke(){}#internal + 95 
    at 120 iosApp                              0x100e7199b        kfun:androidx.compose.runtime#observeDerivedStateRecalculations(kotlin.Function1<androidx.compose.runtime.State<*>,kotlin.Unit>;kotlin.Function1<androidx.compose.runtime.State<*>,kotlin.Unit>;kotlin.Function0<0:0>){0§<kotlin.Any?>} + 1015 
    at 121 iosApp                              0x100f4e933        kfun:androidx.compose.runtime.snapshots.SnapshotStateObserver#observeReads(0:0;kotlin.Function1<0:0,kotlin.Unit>;kotlin.Function0<kotlin.Unit>){0§<kotlin.Any>} + 1155 
    at 122 iosApp                              0x101311ac3        kfun:androidx.compose.ui.node.OwnerSnapshotObserver#observeReads(0:0;kotlin.Function1<0:0,kotlin.Unit>;kotlin.Function0<kotlin.Unit>){0§<androidx.compose.ui.node.OwnerScope>} + 247 
    at 123 iosApp                              0x1013118a7        kfun:androidx.compose.ui.node.OwnerSnapshotObserver#observeMeasureSnapshotReads(androidx.compose.ui.node.LayoutNode;kotlin.Boolean;kotlin.Function0<kotlin.Unit>){} + 443 
    at 124 iosApp                              0x1012d93a7        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.performMeasure#internal + 803 
    at 125 iosApp                              0x1012cd2e7        kfun:androidx.compose.ui.node.LayoutNodeLayoutDelegate.MeasurePassDelegate#remeasure(androidx.compose.ui.unit.Constraints){}kotlin.Boolean + 927 

Process finished with exit code 0

```

**Comment by maltsev-gorskij** [comment]
Looks like fatal is because of this call which uses utils WeekFields.of()
![image](https://github.com/Syer10/compose-material-dialogs/assets/55731252/a770ec90-8a5e-4c3e-b2e0-4cdb07aefd88)

And then we call DayOfWeek() from kotlinx.datetim
![image](https://github.com/Syer10/compose-material-dialogs/assets/55731252/d248c785-5643-448a-b480-36867fc5a487)

And in this library code only allDaysOfWeek[isoDayNumber - 1] with wrong index can throw IndexOutOfBoundsException
![image](https://github.com/Syer10/compose-material-dialogs/assets/55731252/e077f9ca-b832-4513-a435-0630d1fcf784)

So i may conclude thats error are here
![image](https://github.com/Syer10/compose-material-dialogs/assets/55731252/b5fca7f8-16d5-40ce-a428-495e62019a49)

Can you please test my assumption and fix this issue? 

Your library working pretty well, but this ios bug denying me from using it in KMM project.

**Comment by Syer10** [comment]
Found the cause for this and it will be fixed in 0.9.4 👍
